bindings/matlab/+iDynTreeWrappers/prepareVisualization.m

```diff
@@ -70,7 +73,9 @@
 numberOfLinks=length(linkMeshInfo);
 linkNames=cell(numberOfLinks,1);
 figHandles = findobj('Type', 'figure', 'Name', options.name);
-if options.reuseFigure && size(figHandles, 1) > 0
+if strcmp(options.reuseFigure,'gcf')
+    mainHandler=gcf;
```

Comment:
maybe adding `hold off` might solve the issues you saw.

Reply:
I just tried locally. It seems it doesn't unfortunately.

Reply:
Sigh it was worth a try.

Reply:
By further checking, it seems that the ground plane is drawn in any case. In addition, since the ground is drawn with the limits of the figure, every time a bigger ground is drawn since the limits increase. I don't know if there is an easy fix for this. Avoiding to draw the ground if the option if `gcf` would prevent the drawing of the ground also in the first iteration. I guess that the easiest way is for the user to disable the ground from the second iteration on.

Reply:
regarding the `hold off`, I had also tried a few things. As you mentioned, `hold` or setting directly the parameter `NextPlot` to `replace` does not seem to have any effect on the way `patch` draws the meshes on the figure, it always "adds".

But there is another efficient way to deal with this, by using `cla(mainHandler)` instead of `clf(mainHandler,'reset')`. The first keeps the axis configuration, the name and title of the figure. I would actually use it also in the case of the figure match through the 'name', for a coherent behavior.
We either match an existing figure, through its name, or through `gcf`, but in every case, we don't change the existing figure configuration.

I pushed to my fork the suggested code (last commit of branch `visualization/singleFigure-review`), also with the `switch` that could replace the `stcmp` if we use only strings/chars for the parameter `options.reuseFigure`:
https://github.com/S-Dafarra/idyntree/compare/visualization/singleFigure...nunoguedelha:visualization/singleFigure-review

I see that most of those suggestions are already in your code @S-Dafarra .

Reply:
I haven't noticed nor rechecked the problem you experienced with the ground, but it could be there.

Reply:
`cla` is an interesting and better option.

Reply:
To sum-up, the only remaining things I would suggest would be:
- use `cla(mainHandler)` both for 'gcf' and 'name' use cases.
- don't rewrite the name if the field is empty.This should work with all use cases.
- [optional] `switch...case...end` makes it much simpler than `strcmp` to handle use cases based on char arrays (`gcf`,`name`,`none`).

Reply:
> don't rewrite the name if the field is empty.This should work with all use cases.

Actually, here I should check if the `name` option is set.

> use cla(mainHandler) both for 'gcf' and 'name' use cases.

Ok for the 'name' case. But for `gcf`? I mean, isn't it the whole point of using `gcf` the fact that you reuse the current figure as is?

Reply:
`cla(mainHandler)` erases the content of the axes (robot meshes and ground) that has actually to be replaced by the new robot you will draw afterwards.
It does not touch the axes configuration, figure size, name, title and other properties.

Reply:
Regarding `options.name`...
- either you use that field exclusively as a complementary info for the `options.reuseFigure='name'` use case,
- either you use it for setting the name in any use case, as long as `options.name` is not empty.

It's a choice. The important thing is to be able to use `gcf` without worrying about the name of the figure in the caller code using the service.